### ISSUE TITLE
[R package question] Is that possible to retrieve the score for each sample from lgb.cv?

### ISSUE THREAD

**Issue by ajing** [Microsoft/LightGBM#808]
I just wondering whether this is possible. This function can help to analyze the errors during training and do better feature engineering.

**Comment by blewy** [comment]
- Hi ajing, i cant get the lgb.cv to work for modeling a continous variable with regression. Where you able to make it work?  
lgb.train worked just fine with an validation dataset, but gb.cv still puzzles me.

- R gets stuck and the folowing warning shows: 
"Warning messages:
1: In foldVector[y == dimnames(numInClass)$y[i]] <- sample(seqVector) :
  number of items to replace is not a multiple of replacement length "

It seams its searching for a class label. I've used the folowing the code: 

"params <- list(objective = "regression", metric = "l2")
model <- lgb.cv(params,
                dtrain,
                10,
                nfold = 5,
                min_data = 1,
                learning_rate = 1,
                early_stopping_rounds = 10)"

**Comment by ajing** [comment]
What is in your dtrain? What does your y look like?

**Comment by blewy** [comment]
Its a continous variable 

 I created dtrain like this: 
- dtrain <- lgb.Dataset(as.matrix(train.dataset), label = train.target)

and a summary from train.target:
 - summary(train.target)
     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
     0.67   1204.49   2116.89   3032.27   3859.20 106863.00

**Comment by ajing** [comment]
Is train.dataset a data.frame? Does it have any missing value?

**Comment by blewy** [comment]
No Missings and ist a data.frame but I used as.matrix to creata suitable
dataset for lgbm. This thing is that  training with "lgb.train" works fine,
but cv.lgb does not.

Have you used cv.lgb for regression before ? did it worked ?

**Comment by ajing** [comment]
Yes, it worked.

**Comment by blewy** [comment]
Hum.. ok so the problem is on my data set. Thanks.

**Comment by blewy** [comment]
I just tried thos code, its still show warnings :(


"
data<-read.table("./data/lowbwt.dat", header=TRUE, skip=0)

data_train <- data[, c(2:11)]


target <- data_train$BWT
data_train$BWT <- NULL

library(lightgbm)
dtrain <- lgb.Dataset(as.matrix(data_train), label = target)
params <- list(objective = "regression", metric = "l2")
model <- lgb.cv(params,
                dtrain,
                10,
                nfold = 5,
                min_data = 1,
                learning_rate = 1,
                early_stopping_rounds = 10)
"

The data set its the simple birds dataset, lowbwt.dat.

**Comment by ajing** [comment]
Have you try this example:

```
library(lightgbm)
data(agaricus.train, package='lightgbm')
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label=train$label)
params <- list(objective="regression", metric="l2")
model <- lgb.cv(params, dtrain, 10, nfold=5, min_data=1, learning_rate=1, early_stopping_rounds=10)
```

This is a classification problem, but you can use regression to fit the model.

**Comment by blewy** [comment]
Yes, I tried that one, it works fine. But the label (or target variable)
its a 0/1 value. I was just checking the possibility of an issue for true
continuous variables.

**Comment by guolinke** [comment]
duplicated, refer to #283
